### PR TITLE
feat(htmlparser2-adapter): Add CJS build

### PR DESCRIPTION
**Pull request by fb55**
And refactor how CJS is being built.

### CODE REVIEW

File: packages/parse5-htmlparser2-tree-adapter/package.json [hunk 1]
```diff
@@ -14,14 +14,20 @@
         "htmlparser2"
     ],
     "license": "MIT",
-    "main": "dist/index.js",
+    "main": "dist/cjs/index.js",
```
Comment (43081j) [inline packages/parse5-htmlparser2-tree-adapter/package.json:17]: are we sure the default entrypoint should be commonjs? shouldn't we be esm first and cjs second? i.e. only reference the cjs in the exports map

File: packages/parse5-htmlparser2-tree-adapter/package.json [hunk 2]
```diff
@@ -14,14 +14,20 @@
...
     "module": "dist/index.js",
     "types": "dist/index.d.ts",
-    "exports": "./dist/index.js",
+    "exports": {
+        "import": "./dist/index.js",
```
[the remark this answers is not in the document]
Comment (fb55) [reply packages/parse5-htmlparser2-tree-adapter/package.json:21]: This is a bit confusing, agreed. `build:esm` is in the root `package.json` (and make use of project references to build everything at once), the CJS ones are in the package's own `package.json`.

File: packages/parse5-htmlparser2-tree-adapter/package.json (hunk 1 above)
Comment (fb55) [reply packages/parse5-htmlparser2-tree-adapter/package.json:17]: Here is my thinking: The `main` field is only used by legacy NodeJS (everything else either uses the other fields). These versions don't have ESM support, and giving them anything but CJS wouldn't work.

Comment (43081j) [reply packages/parse5-htmlparser2-tree-adapter/package.json:17]: yeah that makes sense. my concern was that tooling that also only looks at `main` would be forced to get CJS. though AFAIK most tooling these days does understand the `module` field or the `exports` field, so we're probably fine 👍

**Pull request merged**